lib/spring/application_manager.rb

```diff
@@ -109,10 +113,24 @@ def start_child(preload = false)
         )
       end
 
-      start_wait_thread(pid, child) if child.gets
+      wait_for_child_to_boot
+      start_wait_thread(pid, child)
+    rescue => e
+      log "error while starting child: #{e.message}"
+      abort("error while starting child: #{e.message}")
+    ensure
       child_socket.close
     end
 
+    def wait_for_child_to_boot
+      timeout = 1
+      loop do
+        break if IO.select([child], nil, nil, timeout)
```

Comment:
In most cases, the child process quickly `puts` something on the socket to indicate that it's ready. Rather than simply wait forever as before (i.e. `child.gets`) we do the same but with a 1 second timeout.